src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -439,32 +439,163 @@ class IndexLogEntryTest extends SparkFunSuite with SQLHelper with BeforeAndAfter
-    val actual1 = aDirectory.merge(bDirectory)
-    val actual2 = bDirectory.merge(aDirectory)
+    val actual1 = directory1.merge(directory2)
+    val actual2 = directory2.merge(directory1)
 
     assert(directoryEquals(actual1, expected))
     assert(directoryEquals(actual2, expected))
+  }
 
-    FileUtils.deleteDirectory(tempDir.toFile)
+  test("Directory.merge: test if merge works as expected when directories overlap.") {
+    // directory1:
+    // a/f1
+    // a/f2
+    // a/b/f3
+    val directory1 = Directory(
+      name = "a",
+      files = Seq(
+        FileInfo("f1", 100L, 100L),
+        FileInfo("f2", 100L, 100L)
+      ),
+      subDirs = Seq(
+        Directory(name = "b", files = Seq(FileInfo("f3", 100L, 100L)))
+      )
+    )
+
+    // directory2:
+    // a/f4
+    // a/b/f5
+    // a/b/f6
+    // a/b/c/f7
+    val directory2 = Directory(
+      name = "a",
+      files = Seq(FileInfo("f4", 100L, 100L)),
+      subDirs = Seq(
+        Directory(
+          name = "b",
+          files = Seq(
+            FileInfo("f5", 100L, 100L),
+            FileInfo("f6", 100L, 100L)
+          ),
+          subDirs = Seq(Directory(
+            name = "c",
+            files = Seq(FileInfo("f7", 100L, 100L))
+          ))
+        )
+      )
+    )
+
+    // Expected result of merging directory1 and directory2:
+    // directory1:
+    // a/f1
+    // a/f2
+    // a/f4
+    // a/b/f3
+    // a/b/f5
+    // a/b/f6
+    // a/b/c/f7
+    val expected = Directory(
+      name = "a",
+      files = Seq(
+        FileInfo("f1", 100L, 100L),
+        FileInfo("f2", 100L, 100L),
+        FileInfo("f4", 100L, 100L)
+      ),
+      subDirs = Seq(
+        Directory(
+          name = "b",
+          files = Seq(
+            FileInfo("f3", 100L, 100L),
+            FileInfo("f5", 100L, 100L),
+            FileInfo("f6", 100L, 100L)
+          ),
+          subDirs = Seq(
+            Directory("c",
+            files = Seq(FileInfo("f7", 100L, 100L)))
+          )
+        )
+      )
+    )
+
+    val actual1 = directory1.merge(directory2)
+    val actual2 = directory2.merge(directory1)
+
+    assert(directoryEquals(actual1, expected))
+    assert(directoryEquals(actual2, expected))
+  }
+
+  test("Directory.merge: test if exception is thrown when directory names are not equal.") {
+    // directory1:
+    // a/f1
+    // a/f2
+    val directory1 = Directory(
+      name = "a",
+      files = Seq(FileInfo("f1", 100L, 100L), FileInfo("f2", 100L, 100L))
+    )
+
+    // directory2:
+    // b/f3
+    // b/f4
+    val directory2 = Directory(
+      name = "b",
+      files = Seq(FileInfo("f3", 100L, 100L), FileInfo("f4", 100L, 100L))
+    )
+
+    val ex1 = intercept[HyperspaceException] (directory1.merge(directory2))
+    val ex2 = intercept[HyperspaceException] (directory2.merge(directory1))
+
+    assert(ex1.msg.contains("Merging directories with names a and b failed."))
```

Comment:
Does it make sense to parametrize `a` and `b` here instead of hard-coding? (for the case if someone changes the name of the directory and forgets to change here).